### PR TITLE
Add rxx gate and cnot to rxx decomposition

### PR DESCRIPTION
**Pull request by chriseclectic**
### Summary

Adds 
* two-qubit rxx rotation gate extension
* `cnot_rxx_decompose` to decompose a CNOT gate into an rxx, rx, and ry rotations.

### CODE REVIEW

File: qiskit/extensions/standard/rxx.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
```
Comment (1ucian0) [inline qiskit/extensions/standard/rxx.py]:
```suggestion
# (C) Copyright IBM 2019.
```

File: qiskit/quantum_info/synthesis/one_qubit_decompose.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
```
Comment (1ucian0) [inline qiskit/quantum_info/synthesis/one_qubit_decompose.py]:
```suggestion
# (C) Copyright IBM 2019.
```

File: qiskit/quantum_info/synthesis/ion_decompose.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
```
Comment (1ucian0) [inline qiskit/quantum_info/synthesis/ion_decompose.py]:
```suggestion
# (C) Copyright IBM 2019.
```

File: qiskit/extensions/standard/rxx.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    """
+
+    def __init__(self, theta):
+        """Create new rxx gate."""
+        super().__init__("rxx", 2, [theta])
+
+    def _define(self):
+        """Calculate a subcircuit that implements this unitary."""
+        from qiskit.quantum_info.synthesis.two_qubit_decompose import two_qubit_cnot_decompose
+        self.definition = two_qubit_cnot_decompose(self.to_matrix())
```
Comment (1ucian0) [inline qiskit/extensions/standard/rxx.py]: I think `self.definition` should be a list.

Comment (chriseclectic) [reply qiskit/extensions/standard/rxx.py]: True, but I'm not sure what the decomposition for this one is...

Comment (kdk) [reply qiskit/extensions/standard/rxx.py]: Can you use the decomposition from #3114 ?

**Pull request merged**